File: program28.java
//Accept n number and display its factors
import java.lang.*;
import java.util.*;

class Factor
{
    public void DisplayFactors(int no)
    {
        for(int i=1;i<=no/2;i++)
        {
            if(no%i==0)
            {
                System.out.println(i);
            }
        }
    }

}


class program28
{
    public static void main(String arg[])
    {
        Factor obj=new Factor();
        int no=0;
        Scanner sobj=new Scanner(System.in);

        System.out.println("Enter number");
        no=sobj.nextInt();

        obj.DisplayFactors(no);
        

    }
}

File: program28.c
//Accept no from user and display its factors
//Input:10
//output1 2 5
#include<stdio.h>

void DisplayFactors(int iNo) //shop
{
    int iCnt=0;
    for(iCnt=1;iCnt<iNo;iCnt++)   //iCnt<=iNo-1
    {
    if((iNo%iCnt)==0)
    {
     printf("%d\n",iCnt);
    }
    
}
}

int main()
{
    auto int iValue=0;

    printf("Enter number:\n");
    scanf("%d",&iValue);

    DisplayFactors(iValue); //home

    return 0;
}


//Time Complexity: O(N).....